### PR TITLE
Run all demos in website, not just components

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

This PR makes it so that all demos tagged in docstrings are run on the website. Previously, we only ran demos for components if they were not listed in the guides. As we add more apis to our docs I think it will be important to automatically run the demos beyond just the components.

### Screenshot of demos run prior to this change
Screen Shot 2022-07-12 at 12 11 51 PM

### Screenshot of demos after this change
Note that `interface_series` and `interface_parallel` are now in the list

image

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/inputs.py [hunk 1]
```diff
@@ -42,7 +42,6 @@ class Number(components.Number):
     """
     Component creates a field for user to enter numeric input. Provides a number as an argument to the wrapped function.
     Input type: float
-    Demos: tax_calculator, titanic_survival
```
Comment (freddyaboulton) [inline gradio/inputs.py:45]: Removing the demos from these deprecated files. We should not be running demos for deprecated components.

File: website/demos/map_demos.py [hunk 2]
```diff
@@ -13,11 +14,16 @@
 port = 7860
 
 demos_to_run = []
-GRADIO_COMPONENTS_FILE = os.path.join(GRADIO_DIR, "gradio", "components.py")
-with open(GRADIO_COMPONENTS_FILE) as comp_file:
-    comp_text = comp_file.read()
-for demostr in re.findall(r'Demos:(.*)', comp_text):
-  demos_to_run += re.findall(r'([a-zA-Z0-9_]+)', demostr)
+for root, _, files in os.walk(pathlib.Path(GRADIO_DIR) / "gradio"):
```
Comment (freddyaboulton) [inline website/demos/map_demos.py:17]: Just applying the same logic as before but on all `.py` files in the repo

File: website/demos/map_demos.py [hunk 3]
```diff
@@ -13,11 +14,16 @@
...
+    for filename in files:
```
Comment (freddyaboulton) [inline website/demos/map_demos.py:18]: Was able to confirm the demos still build correctly on a local docker compose up

image

**Review by aliabid94 — APPROVED** [review]
Makes sense to me. Something we could implement in the future that's even cleaner is to search all the documentation returned from `generate_documentation()` to check demos within the docstrings only.

**Comment by freddyaboulton** [comment]
Completely agree that would be much cleaner @aliabid94 Filed https://github.com/gradio-app/gradio/issues/1783 to track that. Thank you!

**Pull request merged**